src/plugins/discord/address.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import type {NodeAddressT, EdgeAddressT} from "../../core/graph";
+import {NodeAddress, EdgeAddress} from "../../core/graph";
+import {type Snowflake} from "./models";
+import {
+  userNodeType,
+  messageNodeType,
+  sentMessageEdgeType,
+} from "./declaration";
+
+/**
+ * These user IDs are considered globally unique by Discord.
+ * We can reference a user even across multiple guilds using this address.
+ * See https://discordapp.com/api/users/{userId}
+ *
+ * Importantly we don't include any membership details in the address.
+ * This way the graph topology stays identical when users leave / rejoin.
+ */
+export function userAddress(userId: Snowflake): NodeAddressT {
+  return NodeAddress.append(userNodeType.prefix, userId);
+}
+
+/**
+ * The URLs Discord creates to permalink these are:
+ * https://discordapp.com/channels/{guildId}/{channelId}/{messageId}
+ * Meanwhile the API retrieves these using:
+ * https://discordapp.com/api/channels/{channelId}/messages/{messageId}
+ *
+ * In other words the API finds just {channelId} sufficiently namespaced for unique messages.
+ * They're being conservative for the permalinks by namespacing as {guildId}/{channelId}.
+ * As they've clearly not felt the need to change the API, we'll use just {channelId} too.
```

Comment:
One consideration for this might be the case where a SC instance has multiple Discord servers configured and wants to set different weights for each one, which would happen on the guildId level and therefore might warrant including it in the address structure